tests/estimation/test_get_estimation.py

```diff
@@ -93,25 +78,107 @@
 ]
 
 
-@pytest.mark.parametrize("estimator", estimator_list)
-@pytest.mark.parametrize("config", [1, 5])
-@pytest.mark.parametrize("error_tolerance", [TOLERANCE])
-def test_tolerance(estimator, config, error_tolerance):
+parameter_name = [
+    "n",
+    "rg",
+    "mis_spec_m",
+    "mis_spec_y",
+    "dim_x",
+    "dim_m",
+    "seed",
+    "type_m",
+    "sigma_y",
+    "sigma_m",
+    "beta_t_factor",
+    "beta_m_factor",
+]
+
+
+parameter_list = list(
+    itertools.product(
+        [1000],
+        [default_rng(321)],
+        [False],
+        [False],
+        [1, 5],
+        [1],
+        [123],
+        ["binary"],
+        [0.5],
+        [0.5],
+        [0.5],
+        [0.5],
+    )
+)
+
+parameter_list.extend(
+    list(
+        itertools.product(
+            [1000],
+            [default_rng(321)],
+            [False],
+            [False],
+            [1, 5],
+            [1, 5],
+            [123],
+            ["continuous"],
+            [0.5],
+            [0.5],
+            [0.5],
+            [0.5],
+        )
+    )
+)
+
+
+@pytest.fixture(params=parameter_list)
+def dict_param(request):
+    return dict(zip(parameter_name, request.param))
+
+
+@pytest.fixture(autouse=True)
+def data_simulation(dict_param):
+    res = simulate_data(**dict_param)
+    global x, t, m, y, effects
+    x = res[0]
+    t = res[1].ravel()
+    m = res[2]
+    y = res[3].ravel()
+    effects = np.array(res[4:9])
+
+
+@pytest.fixture(params=estimator_list, autouse=True)
+def skip_if_not_implemented(request, dict_param):
+    # config determination
+    if dict_param["dim_m"] == 1 and dict_param["type_m"] == "binary":
+        config = 0
+    else:
+        config = 5
+
+    # try whether estimator is implemented or not
     try:
-        effects_chap = get_estimation(x, t.ravel(), m, y.ravel(), estimator, config)
-        # effects_chap = get_estimation(x, t, m, y, estimator, config)
-        effects_chap = effects_chap[0:5]
-    except Exception as get_estimation_error:
-        if get_estimation_error != NotImplementedError:
+        global effects_chap
+        effects_chap = get_estimation(x, t, m, y, request.param, config)[0:5]
+    except Exception as message_error:
+        if message_error != NotImplementedError:
             pytest.xfail("Missing NotImplementedError")
         else:
             pytest.mark.skip("Not implemented")
-    else:
-        error = abs((effects_chap - effects) / effects)
-        if np.all(np.isnan(effects_chap)):
-            pytest.skip("all effects are NaN")
-        elif np.any(np.isnan(effects_chap)):
-            pprint("NaN found")
-            assert np.all(error[~np.isnan(error)] <= error_tolerance[~np.isnan(error)])
-        else:
-            assert np.all(error <= error_tolerance)
+
+    # NaN situations
+    if np.all(np.isnan(effects_chap)):
+        pytest.skip("all effects are NaN")
+    elif np.any(np.isnan(effects_chap)):
+        pprint("NaN found")
+
+
+def test_tolerance():
```

Comment:
Should take effects as arguments

Reply:
`effects` has been set as a global variable l.160, so test_tolerance() doesn't need arguments.
Is it okay for `effects` to be a global variable?

Reply:
No. Only what you consider as a constant should be a global variable.
The pb is that if people add more testes, they may want to geenrate other data which would make a mess.

Reply:
I fixed it, cf. commit 7784f61.